### ISSUE TITLE
License compatibility of dependencies

### ISSUE THREAD

**Issue by AshwinJay** [fasten-project/fasten#90]
Hello, I see that Fasten has a dependency on the [WebGraph](http://webgraph.di.unimi.it/) library. Fasten is Apache License (ASL) while [WebGraph](https://search.maven.org/artifact/it.unimi.dsi/webgraph) is GPL and this may be an issue:

Ref in the POM: https://github.com/fasten-project/fasten/blob/master/core/pom.xml#L40

As per [ASL](https://www.apache.org/licenses/GPL-compatibility.html):
> Apache 2 software can therefore be included in GPLv3 projects, because the GPLv3 license accepts our software into GPLv3 works. However, GPLv3 software cannot be included in Apache projects. The licenses are incompatible in one direction only, and it is a result of ASF's licensing philosophy and the GPLv3 authors' interpretation of copyright law.
...
We avoid GPLv3 software because merely linking to it is considered by the GPLv3 authors to create a derivative work. We want to honor their license. Unless GPLv3 licensors relax this interpretation of their own license regarding linking, our licensing philosophies are fundamentally incompatible. This is an identical issue for both GPLv2 and GPLv3.

You might want to have this verified. 

Regards.

**Comment by mir-am** [comment]
Thanks for pointing this out. We will definitely look into this possible license conflict. Maybe, @vigna can comment on this.

**Comment by vigna** [comment]
Nope, we modify the licensing exactly to complain with FASTEN's need. It's possible that somewhere it is stated that's GPL. Where did you get that information?
